{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "381\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# reconstruction of early embryonic phylogeny using sypark's script\n",
    "\n",
    "#01\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/51_haplotypecaller\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "for i in temp_list:\n",
    "    if 'HC15-' not in i and '.out' not in i and '.idx' not in i and 'indel' not in i and '.sh' not in i and 'ipyn' not in i and 'snp' in i and 'HC02-01' not in i:\n",
    "        pre_list.append(i)\n",
    "print len(pre_list)\n",
    "#for i in pre_list:\n",
    "#    print i\n",
    "#sys.exit(1)\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/51_run_intersect_381.sh\",\"w\")\n",
    "\n",
    "for input_fn in pre_list:\n",
    "    input_fn = input_fn.split('.')[0]\n",
    "    output_file.write(\"python vcf_intersection.py /home/users/jhyouk/14_colon_LINE1/52_germline_varscan/%s.varscan.snp.vcf /home/users/jhyouk/14_colon_LINE1/51_haplotypecaller/%s.hc.snp.vcf %s VS HC &> %s.ints.log && mv %s.ints.log %s.ints.success\\n\" %(input_fn,input_fn,input_fn,input_fn,input_fn,input_fn))\n",
    "\n",
    "#HC01-21.hc.snp.indel.vcf\n",
    "#HC01-01.varscan.snp.vcf\n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22\n",
      "HC02\n",
      "22\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#02\n",
    "#python extract_variants_in_all_samples.py HC15_23s_common_list.txt HC15_23s_common_ids.txt HC15_subs_in_all_23s.txt 23 &> ext_var.log && mv ext_var.log ext_var.success\n",
    "\n",
    "#id_list = ['HC01','HC02','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "id_list = ['HC02']\n",
    "n=0\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "for i in temp_list:\n",
    "    if 'HC02-' in i and '.VS_HC_common' in i and 'cmn' not in i:\n",
    "        pre_list.append(i)\n",
    "print len(pre_list)\n",
    "#sys.exit(1)\n",
    "\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/52_run_ext_subs_in_all_HC02_22s.sh\",\"w\")\n",
    "for i in id_list:\n",
    "    print i\n",
    "    m=0\n",
    "    output_file1 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_common_list.txt\" %i,\"w\")\n",
    "    output_file2 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_common_ids.txt\" %i,\"w\")\n",
    "\n",
    "    for j in pre_list:\n",
    "        if i in j:\n",
    "            m+=1\n",
    "            input_fn = j.split('.')[0]\n",
    "            output_file1.write(j+'\\n')\n",
    "            output_file2.write(input_fn+'\\n')\n",
    "    output_file1.close()\n",
    "    output_file2.close()\n",
    "    n+=m\n",
    "    output_file.write(\"python extract_variants_in_all_samples.py %s_common_list.txt %s_common_ids.txt %s_subs_in_all.txt %s &> %s_ext_var.log && mv %s_ext_var.log %s_ext_var.success\\n\" %(i,i,i,str(m),i,i,i))\n",
    "print n    \n",
    "    \n",
    "\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#03\n",
    "#python remove_common_variants.py HC15_23s_common_list.txt HC15_subs_in_all_23s.txt &> rm_cmn.log && mv rm_cmn.log rm_cmn.success\n",
    "\n",
    "id_list = ['HC01','HC02','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/53_remove_common_sub_381.sh\",\"w\")\n",
    "for i in id_list:\n",
    "    output_file.write(\"python remove_common_variants.py %s_common_list.txt %s_subs_in_all.txt &> %s_rm_cmn.log && mv %s_rm_cmn.log %s_rm_cmn.success\\n\" %(i,i,i,i,i))\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22\n",
      "405\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#04\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "for i in temp_list:\n",
    "    if 'HC02-' in i and '.cmn_fi' in i and 'readinfo.readc.rasm2' not in i:\n",
    "        pre_list.append(i)\n",
    "print len(pre_list)\n",
    "#for i in pre_list:\n",
    "#    print i\n",
    "#sys.exit(1)\n",
    "\n",
    "info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "info_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print(len(info_dic))\n",
    "\n",
    "\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/54_run_readinfo_annot_HC02_22s.sh\",\"w\")\n",
    "\n",
    "for input_fn in pre_list:\n",
    "    input_fn = input_fn.split('.')[0]\n",
    "    output_file.write(\"python point_mutation_annotation_serial_run.py -i %s.VS_HC_common.cmn_fi -t /home/users/team_projects/colon_LINE1/02_bam/%s\\\n",
    " -n /home/users/team_projects/colon_LINE1/02_bam/HC15_germline_30x.s.md.bam &> %s.ri_annot.log &&\\\n",
    " mv %s.ri_annot.log %s.ri_annot.success\\n\" %(input_fn,info_dic[input_fn],input_fn,input_fn,input_fn))\n",
    "\n",
    "\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#python point_mutation_annotation_serial_run.py -i 10_ALL13-4_006H8.VS_HC_common.cmn_fi -t /home/users/team_projects/Lineage_tracing/DB10/02_BAM/10_ALL13-4_006H8.s.md.ir.br.bam -n /home/users/team_projects/Lineage_tracing/DB2/02_BAM/2_blood_3s_merged.s.md.ir.br.bam &> 10_ALL13-4_006H8.ri_annot.log && mv 10_ALL13-4_006H8.ri_annot.log 10_ALL13-4_006H8.ri_annot.success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#05\n",
    "#id_list = ['HC01','HC15','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "id_list = ['HC02']\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "for i in temp_list:\n",
    "    if 'HC02-' in i and 'rasm2' in i:\n",
    "        pre_list.append(i)\n",
    "print len(pre_list)\n",
    "#for i in pre_list:\n",
    "#    print i\n",
    "#sys.exit(1)\n",
    "\n",
    "#info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "#info_line = info_file.readline().strip()\n",
    "#info_line = info_file.readline().strip()\n",
    "#\n",
    "#info_dic = {}\n",
    "#while info_line:\n",
    "#    info_split = info_line.split('\\t')\n",
    "#    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "#    info_line = info_file.readline().strip()\n",
    "\n",
    "#print(len(info_dic))\n",
    "\n",
    "\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/55_run_np_annot_HC02_22s.sh\",\"w\")\n",
    "\n",
    "for input_fn in pre_list:\n",
    "    input_fn = input_fn.split('.')[0]\n",
    "    output_file.write(\"sh ./SNV_NormalPanel_Serial_Annotation_modified.sh %s.VS_HC_common.cmn_fi.readinfo.readc.rasm2 \\\n",
    "&> %s.np_annot.log && mv %s.np_annot.log %s.np_annot.success\\n\" %(input_fn,input_fn,input_fn,input_fn))\n",
    "\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "383\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#06\n",
    "\n",
    "id_list = ['HC01','HC15','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "#id_list = ['HC02']\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "for i in temp_list:\n",
    "    if 'HC02-' not in i and '.snuN30.bgiN24' in i:\n",
    "        pre_list.append(i)\n",
    "print len(pre_list)\n",
    "#for i in pre_list:\n",
    "#    print i\n",
    "#sys.exit(1)\n",
    "\n",
    "#info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "#info_line = info_file.readline().strip()\n",
    "#info_line = info_file.readline().strip()\n",
    "#\n",
    "#info_dic = {}\n",
    "#while info_line:\n",
    "#    info_split = info_line.split('\\t')\n",
    "#    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "#    info_line = info_file.readline().strip()\n",
    "\n",
    "#print(len(info_dic))\n",
    "\n",
    "\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/56_run_anv_383s.sh\",\"w\")\n",
    "\n",
    "for input_fn in pre_list:\n",
    "    input_id = input_fn.split('.')[0]\n",
    "    output_file.write(\"python annotate_using_annovar_v2.py %s -i 'refGene,1000g2015aug_all,1000g2015aug_eas,exac03,esp6500siv2_all,\\\n",
    "avsnp150,cosmic86_coding' &> %s.anv.log && mv %s.anv.log %s.anv.success\\n\" %(input_fn,input_id,input_id,input_id))\n",
    "\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "383\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#07\n",
    "id_list = ['HC01','HC15','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "#id_list = ['HC02']\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "for i in temp_list:\n",
    "    if 'HC02-' not in i and '.VS_HC_common.cmn_fi.readinfo.readc.rasm2.snuN30.bgiN24.anv' in i and '.fi' not in i:\n",
    "        pre_list.append(i)\n",
    "print len(pre_list)\n",
    "#for i in pre_list:\n",
    "#    print i\n",
    "#sys.exit(1)\n",
    "\n",
    "#info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "#info_line = info_file.readline().strip()\n",
    "#info_line = info_file.readline().strip()\n",
    "#\n",
    "#info_dic = {}\n",
    "#while info_line:\n",
    "#    info_split = info_line.split('\\t')\n",
    "#    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "#    info_line = info_file.readline().strip()\n",
    "\n",
    "#print(len(info_dic))\n",
    "\n",
    "\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/57_run_1st_filter_383s_1.sh\",\"w\")\n",
    "\n",
    "n = 0\n",
    "m=1\n",
    "for input_fn in pre_list:\n",
    "    n+=1\n",
    "    if n % 30 == 0:\n",
    "        m+=1\n",
    "        output_file.close()\n",
    "        output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/57_run_1st_filter_383s_%s.sh\" % str(m),\"w\")\n",
    "\n",
    "        \n",
    "    input_id = input_fn.split('.')[0]\n",
    "    output_file.write(\"python DB_1st_filter_standard.py %s 151\\n\" %(input_fn))\n",
    "\n",
    "    \n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "383\n",
      "HC01\n",
      "HC15\n",
      "HC03\n",
      "HC04\n",
      "HC05\n",
      "HC06\n",
      "HC08\n",
      "HC09\n",
      "HC10\n",
      "HC12\n",
      "HC13\n",
      "HC14\n",
      "HC16\n",
      "HC17\n",
      "HC18\n",
      "HC19\n",
      "HC20\n",
      "HC21\n",
      "383\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#08\n",
    "\n",
    "id_list = ['HC01','HC15','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "#id_list = ['HC02']\n",
    "n=0\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "for i in temp_list:\n",
    "    if 'HC02-' not in i and '.VS_HC_common.cmn_fi.readinfo.readc.rasm2.snuN30.bgiN24.anv.fi' in i:\n",
    "        pre_list.append(i)\n",
    "print len(pre_list)\n",
    "#sys.exit(1)\n",
    "\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/58_run_merge_list_383s.sh\",\"w\")\n",
    "for i in id_list:\n",
    "    print i\n",
    "    m=0\n",
    "    output_file1 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_merge_input_list.txt\" %i,\"w\")\n",
    "    output_file2 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_merge_input_ids.txt\" %i,\"w\")\n",
    "\n",
    "    for j in pre_list:\n",
    "        if i in j:\n",
    "            m+=1\n",
    "            input_fn = j.split('.')[0]\n",
    "            output_file1.write(j+'\\n')\n",
    "            output_file2.write(input_fn+'\\n')\n",
    "    output_file1.close()\n",
    "    output_file2.close()\n",
    "    n+=m\n",
    "    output_file.write(\"python vcfs_merge_casenum_caseid_fast.py %s_merge_input_list.txt %s_merge_input_ids.txt %s_merged.txt\\n\" %(i,i,i))\n",
    "print n    \n",
    "    \n",
    "\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "HC01\n",
      "23\n",
      "HC15\n",
      "23\n",
      "HC03\n",
      "24\n",
      "HC04\n",
      "23\n",
      "HC05\n",
      "24\n",
      "HC06\n",
      "22\n",
      "HC08\n",
      "23\n",
      "HC09\n",
      "21\n",
      "HC10\n",
      "20\n",
      "HC12\n",
      "22\n",
      "HC13\n",
      "22\n",
      "HC14\n",
      "19\n",
      "HC16\n",
      "18\n",
      "HC17\n",
      "14\n",
      "HC18\n",
      "17\n",
      "HC19\n",
      "23\n",
      "HC20\n",
      "22\n",
      "HC21\n",
      "23\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "## Extra. production of merged_mpu per each patient\n",
    "import os,sys\n",
    "id_list = ['HC01','HC15','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "#id_list = ['HC02']\n",
    "\n",
    "info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "info_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print(len(info_dic))\n",
    "\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/03_mpileup/merged_mpileup/01_run.383.mpu.call.sh\",\"w\")\n",
    "\n",
    "for pt_num in id_list:\n",
    "    print pt_num\n",
    "    crypt_num = 0\n",
    "    temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark\")\n",
    "    file_list = []\n",
    "    pre_list = []\n",
    "    for i in temp_list:\n",
    "        if pt_num in i and '.VS_HC_common' in i and '.cmn_fi' not in i:\n",
    "            pre_list.append(i)\n",
    "    print len(pre_list)\n",
    "    crypt_num = str(len(pre_list))\n",
    "    #sys.exit(1)\n",
    "    #for i in pre_list:\n",
    "    #    print i\n",
    "\n",
    "    #output_file = file(\"/home/users/jhyouk/14_colon_LINE1/03_mpileup/merged_mpileup/01_run.%s.%s.mpu.call.sh\" %(pt_num,crypt_num),\"w\")\n",
    "    bam_list = ''\n",
    "    for input_fn in pre_list:\n",
    "        input_id = input_fn.split('.')[0]\n",
    "        bam_list = bam_list + '/home/users/team_projects/colon_LINE1/02_bam/'+ info_dic[input_id] + ' '\n",
    "    #print bam_list\n",
    "    #sys.exit(1)\n",
    "\n",
    "    for ii in range(1,26):\n",
    "        chr_num = str(ii).replace('23','X').replace(\"24\",\"Y\").replace(\"25\",\"MT\")\n",
    "        output_file.write(\"(samtools mpileup -AB -d 10000 -q 20 -Q 20 -f /home/users/data/01_reference/human_g1k_v37/human_g1k_v37.fasta -r \\\n",
    "%s %s| gzip > %s.%ss.q20Q20.chr%s.mpileup.gz) 2> %s.mpu.chr%s.log && mv %s.mpu.chr%s.log %s.mpu.chr%s.success \\\n",
    "&& python /home/users/sypark/01_Python_files/normal_matrix/pileup_calling_snv.py %s.%ss.q20Q20.chr%s.mpileup.gz %s \\\n",
    "&> %s.chr%s.snv_call.log && mv %s.chr%s.snv_call.log %s.chr%s.snv_call.success && \\\n",
    "python /home/users/sypark/01_Python_files/normal_matrix/pileup_calling_indel_190705.py %s.%ss.q20Q20.chr%s.mpileup.gz %s \\\n",
    "&> %s.chr%s.indel_call.log && mv %s.chr%s.indel_call.log %s.chr%s.indel_call.success\\n\" \\\n",
    "%(chr_num,bam_list,pt_num,crypt_num,chr_num,pt_num,chr_num,pt_num,chr_num,pt_num,chr_num,\\\n",
    "pt_num,crypt_num,chr_num,crypt_num,\\\n",
    "pt_num,chr_num,pt_num,chr_num,pt_num,chr_num,\\\n",
    "pt_num,crypt_num,chr_num,crypt_num,\\\n",
    "pt_num,chr_num,pt_num,chr_num,pt_num,chr_num))\n",
    "\n",
    "    #(samtools mpileup -AB -d 10000 -q 20 -Q 20 -f /home/users/data/01_reference/human_g1k_v37/human_g1k_v37.fasta -r %s \\\n",
    "    #10_ALL13-4_006H8.s.md.ir.br.bam 10_ALL13-8_001G5.s.md.ir.br.bam 10_ALL17-3_001D3.s.md.ir.br.bam 10_ALL17-5_002B3.s.md.ir.br.bam 10_ALL19-2_001B5.s.md.ir.br.bam 10_ARL10-4_001D4-B6.s.md.ir.br.bam 10_ARL10-4_001D4.s.md.ir.br.bam 10_ARL12-1_001B3.s.md.ir.br.bam 10_ARL12-2_001C12.s.md.ir.br.bam 10_ARL12-4_001H11.s.md.ir.br.bam 10_ARL1-3_003G11.s.md.ir.br.bam 10_ARL13-1_001G8.s.md.ir.br.bam 10_ARL1-5_003B3.s.md.ir.br.bam 10_ARL23-6_002E11.s.md.ir.br.bam 10_ARL3-2_006B8.s.md.ir.br.bam 10_ARL6-1_001A9.s.md.ir.br.bam 10_ARL6-2_001E7.s.md.ir.br.bam 10_ARL7-2_005B2.s.md.ir.br.bam 10_ARL8-4_001D12.s.md.ir.br.bam 10_ARL9-2_005G8.s.md.ir.br.bam 10_ARL9-3_001D5.s.md.ir.br.bam 10_ARLFb2-5_002G8.s.md.ir.br.bam 10_Blood_3s_merged.s.md.ir.br.bam 10_LA14-2_001E7.s.md.ir.br.bam 10_LA14-2_002D1-A5.s.md.ir.br.bam 10_LA14-2_002D1.s.md.ir.br.bam 10_LA15-2_001D7.s.md.ir.br.bam 10_LAFb11-3_001G8.s.md.ir.br.bam 10_LAFb16-2_001D8.s.md.ir.br.bam 10_LAFb16-3_001E11.s.md.ir.br.bam 10_LAFb8-2_001D4.s.md.ir.br.bam 10_PLL2-2_001F4.s.md.ir.br.bam 10_PLL4-4_003F8.s.md.ir.br.bam 10_PLL6-3_002A11.s.md.ir.br.bam 10_PLL6-3_002D6.s.md.ir.br.bam 10_PRL10-2_001G10.s.md.ir.br.bam 10_PRL10-3_001G7.s.md.ir.br.bam 10_PRL10-5_001E2.s.md.ir.br.bam 10_PRL12-5_001C5.s.md.ir.br.bam 10_PRL16-1_001E6.s.md.ir.br.bam 10_PRL2-4_002C4.s.md.ir.br.bam 10_PRL3-1_001B8-E10.s.md.ir.br.bam 10_PRL3-1_001B8.s.md.ir.br.bam 10_PRL3-1_001E4.s.md.ir.br.bam 10_PRL3-2_002E11.s.md.ir.br.bam 10_PRL3-3_001B11.s.md.ir.br.bam 10_PRL4-1_004C1.s.md.ir.br.bam 10_PRL4-4_001E8.s.md.ir.br.bam 10_PRLM2-3_001D10.s.md.ir.br.bam 10_RA9-3_005G11.s.md.ir.br.bam 10_RA9-4_005B8.s.md.ir.br.bam 10_UO3-2_001C6.s.md.ir.br.bam\n",
    "    # | gzip > DB10.52s.q20Q20.chr5.mpileup.gz) 2> DB10.mpu.chr5.log && mv DB10.mpu.chr5.log DB10.mpu.chr5.success && python /home/users/sypark/01_Python_files/normal_matrix/pileup_calling_snv.py DB10.52s.q20Q20.chr5.mpileup.gz 52 &> DB10.chr5.snv_call.log && mv DB10.chr5.snv_call.log DB10.chr5.snv_call.success && python /home/users/sypark/01_Python_files/normal_matrix/pileup_calling_indel_190705.py DB10.52s.q20Q20.chr5.mpileup.gz 52 &> DB10.chr5.indel_call.log && mv DB10.chr5.indel_call.log DB10.chr5.indel_call.success\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "HC14\n",
      "HC15\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#09\n",
    "info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "info_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print(len(info_dic))\n",
    "\n",
    "\n",
    "\n",
    "#id_list = ['HC01','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "id_list = ['HC14','HC15']\n",
    "\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/59_run_sampleN_annot_HC1415s.sh\",\"w\")\n",
    "for i in id_list:\n",
    "    print i\n",
    "    n=0\n",
    "    for j in info_dic.keys():\n",
    "        if i in j:\n",
    "            n+=1\n",
    "            \n",
    "    output_file.write(\"python DB_AddSampleNinfo.py %s_merged.txt /home/users/jhyouk/14_colon_LINE1/03_mpileup/HC14_15_merged_mpileup/\\\n",
    "%s.%ss.q20Q20.chr1.mpileup.snv.%sN.gz &> %s_sampleN_annot.log && mv %s_sampleN_annot.log %s_sampleN_annot.success\\n\" %(i,i,str(n),str(n),i,i,i))\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "#python DB_AddSampleNinfo.py DB10_50s_merged.txt /home/users/sypark/00_Project/06_LineageTracing/db10/02_mpileup/01_merged_mpu_q20Q20/DB10.52s.q20Q20.chr1.mpileup.snv.50N.gz &> sampleN_annot.log && mv sampleN_annot.log sampleN_annot.success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "HC14\n",
      "19\n",
      "HC14\n",
      "HC15\n",
      "23\n",
      "HC15\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "## Extra2\n",
    "\n",
    "import os,sys\n",
    "#id_list = ['HC01','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "id_list = ['HC14','HC15']\n",
    "\n",
    "info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "info_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print(len(info_dic))\n",
    "\n",
    "for pt_num in id_list:\n",
    "    print pt_num\n",
    "    crypt_num = 0\n",
    "    temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark\")\n",
    "    file_list = []\n",
    "    pre_list = []\n",
    "    for i in temp_list:\n",
    "        if pt_num in i and '.VS_HC_common' in i and '.cmn_fi' not in i:\n",
    "            pre_list.append(i)\n",
    "    print len(pre_list)\n",
    "    crypt_num = str(len(pre_list))\n",
    "    #sys.exit(1)\n",
    "    #for i in pre_list:\n",
    "    #    print i\n",
    "\n",
    "    #output_file = file(\"/home/users/jhyouk/14_colon_LINE1/03_mpileup/merged_mpileup/01_run.%s.%s.mpu.call.sh\" %(pt_num,crypt_num),\"w\")\n",
    "    bam_list = ''\n",
    "    for input_fn in pre_list:\n",
    "        input_id = input_fn.split('.')[0]\n",
    "        bam_list = bam_list + '/home/users/team_projects/colon_LINE1/02_bam/'+ info_dic[input_id] + ' '\n",
    "    #print bam_list\n",
    "    #sys.exit(1)\n",
    "    i=pt_num\n",
    "    output_file = file(\"/home/users/jhyouk/14_colon_LINE1/03_mpileup/HC14_15_merged_mpileup/02_run_snv_mpu_counting_%s.sh\" %i,\"w\")\n",
    "    print i\n",
    "    for j in range(1,26):\n",
    "        j = str(j).replace(\"23\",\"X\").replace(\"24\",\"Y\").replace(\"25\",\"MT\")\n",
    "        output_file.write(\"python %s_mpu_snv_sampleN_Dpmore10.py %s.%ss.q20Q20.chr%s.mpileup.snv.gz &> %s_chr%s.snv_count.log && \\\n",
    "mv %s_chr%s.snv_count.log %s_chr%s.snv_count.success\\n\" % (i,i,len(pre_list),str(j),i,str(j),i,str(j),i,str(j)))\n",
    "\n",
    "    output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19\n",
      "['HC14-05', 'HC14-13', 'HC14-08', 'HC14-06', 'HC14-16', 'HC14-11', 'HC14-09', 'HC14-03', 'HC14-02', 'HC14-10', 'HC14-19', 'HC14-07', 'HC14-12', 'HC14-14', 'HC14-01', 'HC14-04', 'HC14-17', 'HC14-15', 'HC14-18']\n"
     ]
    }
   ],
   "source": [
    "#Extra3\n",
    "temp = '/home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-05-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-13-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-08-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-06-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-16-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-11-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-09-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-03-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-02-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-10-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-19-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-07-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-12-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-14-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-01-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-04-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-17-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-15-wgs-ILLUMINA.fmarked.bam /home/users/team_projects/colon_LINE1/02_bam/JHY-COL-HC14-18-wgs-ILLUMINA.fmarked.bam'\n",
    "a = temp.split(' ')\n",
    "b=[]\n",
    "for i in a:\n",
    "    b.append(i.split('/')[-1].replace(\"JHY-COL-\",\"\").replace(\"-wgs-ILLUMINA.fmarked.bam\",\"\").replace(\".s.md.bam\",\"\"))\n",
    "    \n",
    "print len(b)\n",
    "print b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Extra4 sequenza segment cleaning\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "n=0\n",
    "for i in temp_list:\n",
    "    if \".100kbcov\" not in i and \".gz\" not in i and \".R\" not in i and \".sh\" not in i and \".py\" not in i and '.ipynb' not in i:\n",
    "        if \"qsub\" not in i:\n",
    "            file_list.append(i)\n",
    "            #print i\n",
    "print len(file_list)\n",
    "#sys.exit(1)\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/51_run_segment_clean.sh\",\"w\")\n",
    "for j in file_list:\n",
    "    output_file.write(\"python seqz_segment_clear.py /home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/%s/%s_segments.txt &> %s.out \\n\" % (j,j,j))\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "19\n",
      "23\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#10\n",
    "import os,sys\n",
    "#id_list = ['HC01','HC15','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "id_list = ['HC14','HC15']\n",
    "gender_list = ['F','F']\n",
    "\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza\")\n",
    "file_list = []\n",
    "\n",
    "n=0\n",
    "for i in temp_list:\n",
    "    if \".100kbcov\" not in i and \".gz\" not in i and \".R\" not in i and \".sh\" not in i and \".py\" not in i and '.ipynb' not in i:\n",
    "        if \"qsub\" not in i:\n",
    "            file_list.append(i)\n",
    "            #print i\n",
    "print len(file_list)\n",
    "#sys.exit(1)\n",
    "\n",
    "n=0\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/60_run_loh_annot_HC1415.sh\",\"w\")\n",
    "for id_fn in id_list:\n",
    "    pre_list = []\n",
    "    for j in file_list:\n",
    "        if id_fn in j:\n",
    "            pre_list.append(j)\n",
    "    print len(pre_list)\n",
    "    #sys.exit(1)\n",
    "\n",
    "    # generate seqz_clean_list.txt\n",
    "    output_file1 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_seqz_clean_list.txt\" %id_fn,\"w\")\n",
    "    for ii in pre_list:\n",
    "        output_file1.write(\"%s\\t/home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/%s/%s_segments.txt.clean\\n\" %(ii.replace(\"JHY-COL-\",\"\").replace(\"_blood-wgs-ILLUMINA.fmarked\",\"\"),ii,ii))   \n",
    "    output_file1.close()\n",
    "    output_file.write(\"python add_loh_samples.py %s_merged.txt.sampleN %s_seqz_clean_list.txt %s &> %s.60_run.out\\n\" %(id_fn,id_fn,gender_list[id_list.index(id_fn)],id_fn) )\n",
    "    \n",
    "    n+=1\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'HC02'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-128-fce797d14fd5>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0ma\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0ma\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'HC02'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'1'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m3\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0;32mprint\u001b[0m \u001b[0ma\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'HC02'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'1'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'HC02'"
     ]
    }
   ],
   "source": [
    "a={}\n",
    "a['HC02']['1'] = 3\n",
    "print a['HC02']['1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "HC01\n",
      "HC03\n",
      "HC04\n",
      "HC05\n",
      "HC06\n",
      "HC08\n",
      "HC09\n",
      "HC10\n",
      "HC12\n",
      "HC13\n",
      "HC14\n",
      "HC15\n",
      "HC16\n",
      "HC17\n",
      "HC18\n",
      "HC19\n",
      "HC20\n",
      "HC21\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Extra\n",
    "#sequenza plot reconstruction with gender information\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza\")\n",
    "file_list = []\n",
    "\n",
    "n=0\n",
    "for i in temp_list:\n",
    "    if \".100kbcov\" not in i and \".gz\" not in i and \".R\" not in i and \".sh\" not in i and \".py\" not in i and '.ipynb' not in i:\n",
    "        if \"qsub\" not in i:\n",
    "            file_list.append(i)\n",
    "print len(file_list)\n",
    "#sys.exit(1)\n",
    "            \n",
    "            \n",
    "id_list = ['HC01','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC15','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "gender_list = ['M','M','M','F','M','F','M','M','F','M','F','F','F','F','F','M','F','F']\n",
    "#print len(id_list),len(gender_list)\n",
    "\n",
    "#id_list = ['HC02']\n",
    "#gender_list = ['M']\n",
    "output_file=file(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/01_run_pdf_gender_383.sh\",\"w\")\n",
    "for id_fn in id_list:\n",
    "    print id_fn\n",
    "    k = gender_list[id_list.index(id_fn)]\n",
    "    if k=='M':\n",
    "        kk = 'XY'\n",
    "    else:\n",
    "        kk='XX'\n",
    "    for j in file_list:\n",
    "        if id_fn in j:\n",
    "            if 'JHY-COL' in j:\n",
    "                output_file.write(\"/home/users/kjyi/miniconda3/bin/Rscript ../04_sequenza/11_Rscript_sequenza_6args.R ../04_sequenza/%s.comp.seqz.rmGLMT.gz %s 1 2 hg19 %s &> %s.Rscript_gender.out\\n\" %(j,j,kk,j))\n",
    "            else:\n",
    "                output_file.write(\"/home/users/kjyi/miniconda3/bin/Rscript ../04_sequenza/11_Rscript_sequenza_6args.R ../04_sequenza/%s.comp.seqz.rmGLMTJH.gz %s 1 2 hg19 %s &> %s.Rscript_gender.out\\n\" %(j,j,kk,j))\n",
    "                \n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#11\n",
    "info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "info_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print(len(info_dic))\n",
    "\n",
    "#id_list = ['HC01','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC15','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "id_list = ['HC14','HC15']\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/61_run_merged_germ_fi_HC1415.sh\",\"w\")\n",
    "\n",
    "for id_fn in id_list:\n",
    "    output_file1 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_merge_input_ids.txt\" % id_fn,\"w\")\n",
    "    output_file2 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_excluded_ids.txt\" % id_fn,\"w\")\n",
    "    for j in info_dic.keys():\n",
    "        if id_fn in j:\n",
    "            output_file1.write(j+'\\n')\n",
    "    \n",
    "    output_file.write(\"python merged_germ_filter.py %s_merged.txt.sampleN.loh_id %s_merge_input_ids.txt %s_excluded_ids.txt\\n\" %(id_fn,id_fn,id_fn))\n",
    "    output_file1.close()\n",
    "    output_file2.close()\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "n= 19\n",
      "n= 23\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#12\n",
    "info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "info_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print(len(info_dic))\n",
    "\n",
    "#id_list = ['HC01','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC15','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "id_list = ['HC14','HC15']\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/62_run_sampling_for_HM_HC1415.sh\",\"w\")\n",
    "\n",
    "for id_fn in id_list:\n",
    "    n=0\n",
    "    for j in info_dic.keys():\n",
    "        if id_fn in j:\n",
    "            n+=1\n",
    "    print 'n=',n\n",
    "    output_file.write(\"python shared_mt_sampling.py %s_merged.txt.sampleN.loh_id.g_fi %s 3\\n\" %(id_fn,str(n)))\n",
    "    output_file.write(\"python shared_mt_sampling.py %s_merged.txt.sampleN.loh_id.g_fi %s 5\\n\" %(id_fn,str(n)))\n",
    "    output_file.write(\"python shared_mt_sampling.py %s_merged.txt.sampleN.loh_id.g_fi %s 2\\n\" %(id_fn,str(n)))\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "n= 19\n",
      "n= 23\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#13\n",
    "info_file = file(\"/home/users/jhyouk/14_colon_LINE1/colon_LINE1_405_metadata_v2.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "info_dic = {}\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_dic[info_split[0]] = info_split[1]\n",
    "\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print(len(info_dic))\n",
    "\n",
    "\n",
    "#id_list = ['HC01','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC15','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "id_list = ['HC14','HC15']\n",
    "output_file = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/63_run_multibam_mpu_for_HM_HC1415.sh\",\"w\")\n",
    "\n",
    "for id_fn in id_list:\n",
    "    output_file1 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_bam_list.txt\" % id_fn,\"w\")\n",
    "    output_file2 = file(\"/home/users/jhyouk/14_colon_LINE1/53_early_embryonic_tree_bySYPark/%s_bam_id_list.txt\" % id_fn,\"w\")\n",
    "    n=0\n",
    "    for j in info_dic.keys():\n",
    "        if id_fn in j:\n",
    "            n+=1\n",
    "            output_file1.write('/home/users/team_projects/colon_LINE1/02_bam/'+info_dic[j]+'\\n')\n",
    "            output_file2.write(j+'\\n')\n",
    "    print 'n=',n\n",
    "    output_file1.close()\n",
    "    output_file2.close()\n",
    "    output_file.write(\"python DB_mpu_multibam_call_edit.py %s_merged.txt.sampleN.loh_id.g_fi.sam2.txt %s_bam_list.txt %s_bam_id_list.txt %s &> %s_mb_mpu.log && mv %s_mb_mpu.log %s_mb_mpu.success\\n\" %(id_fn,id_fn,id_fn,str(n),id_fn,id_fn,id_fn))\n",
    "    output_file.write(\"python DB_mpu_multibam_call_edit.py %s_merged.txt.sampleN.loh_id.g_fi.sam3.txt %s_bam_list.txt %s_bam_id_list.txt %s &> %s_mb_mpu.log && mv %s_mb_mpu.log %s_mb_mpu.success\\n\" %(id_fn,id_fn,id_fn,str(n),id_fn,id_fn,id_fn))\n",
    "    output_file.write(\"python DB_mpu_multibam_call_edit.py %s_merged.txt.sampleN.loh_id.g_fi.sam5.txt %s_bam_list.txt %s_bam_id_list.txt %s &> %s_mb_mpu.log && mv %s_mb_mpu.log %s_mb_mpu.success\\n\" %(id_fn,id_fn,id_fn,str(n),id_fn,id_fn,id_fn))\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "HC05-01\n",
      "8\t79078613\t80305754\t2\t2\t0\n",
      "JHY-COL-HC06-20_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28806428\t59033610\t3\t1\t0\n",
      "JHY-COL-HC13-13_blood-wgs-ILLUMINA.fmarked\n",
      "2\t90532523\t92325966\t0\t0\t0\n",
      "JHY-COL-HC13-13_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28784163\t59033357\t2\t1\t0\n",
      "HC19-09\n",
      "Y\t28790987\t59033375\t2\t1\t0\n",
      "JHY-COL-HC10-03_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28802898\t59033610\t3\t1\t0\n",
      "JHY-COL-HC13-18_blood-wgs-ILLUMINA.fmarked\n",
      "2\t89876953\t92325966\t3\t2\t1\n",
      "JHY-COL-HC06-18_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t23901106\t59033610\t2\t1\t0\n",
      "JHY-COL-HC10-08_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28549246\t59033610\t2\t1\t0\n",
      "JHY-COL-HC04-21_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28810391\t59033610\t2\t1\t0\n",
      "JHY-COL-HC04-04_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28812934\t59033610\t3\t1\t0\n",
      "JHY-COL-HC08-16_blood-wgs-ILLUMINA.fmarked\n",
      "2\t90381963\t92325966\t4\t3\t1\n",
      "JHY-COL-HC19-20_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28700764\t59033375\t2\t1\t0\n",
      "JHY-COL-HC10-12_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28818558\t59033610\t2\t1\t0\n",
      "JHY-COL-HC10-17_blood-wgs-ILLUMINA.fmarked\n",
      "2\t89285734\t92325966\t3\t2\t1\n",
      "JHY-COL-HC10-17_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t23900125\t59033610\t2\t1\t0\n",
      "JHY-COL-HC10-18_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t23897532\t59033610\t2\t1\t0\n",
      "JHY-COL-HC19-16_blood-wgs-ILLUMINA.fmarked\n",
      "6\t57369086\t58779898\t4\t2\t2\n",
      "JHY-COL-HC19-02_blood-wgs-ILLUMINA.fmarked\n",
      "6\t57473099\t58779898\t4\t2\t2\n",
      "HC19-19\n",
      "Y\t28790604\t59033375\t3\t1\t0\n",
      "JHY-COL-HC06-12_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28802921\t59033610\t3\t1\t0\n",
      "JHY-COL-HC10-11_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28797308\t59033610\t2\t1\t0\n",
      "JHY-COL-HC10-20_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t23898898\t59033610\t2\t1\t0\n",
      "JHY-COL-HC04-18_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28801616\t59033610\t2\t1\t0\n",
      "JHY-COL-HC19-04_blood-wgs-ILLUMINA.fmarked\n",
      "6\t57369480\t58779898\t4\t2\t2\n",
      "JHY-COL-HC18-05_blood-wgs-ILLUMINA.fmarked\n",
      "8\t46839286\t48147916\t3\t2\t1\n",
      "JHY-COL-HC19-23_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28792268\t59033375\t2\t1\t0\n",
      "JHY-COL-HC09-21_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28814922\t59033610\t4\t1\t0\n",
      "JHY-COL-HC02-16_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28819200\t59033375\t6\t1\t0\n",
      "JHY-COL-HC05-15_blood-wgs-ILLUMINA.fmarked\n",
      "11\t113412\t46594382\t2\t2\t0\n",
      "JHY-COL-HC13-02_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28818256\t59033357\t4\t1\t0\n",
      "JHY-COL-HC14-07_blood-wgs-ILLUMINA.fmarked\n",
      "15\t62102252\t68690152\t4\t2\t2\n",
      "JHY-COL-HC12-23_blood-wgs-ILLUMINA.fmarked\n",
      "2\t89872282\t92325966\t2\t2\t0\n",
      "JHY-COL-HC02-21_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t22268919\t59033375\t2\t1\t0\n",
      "JHY-COL-HC09-18_blood-wgs-ILLUMINA.fmarked\n",
      "10\t62646486\t64649521\t2\t2\t0\n",
      "JHY-COL-HC13-05_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28790409\t59033357\t2\t1\t0\n",
      "JHY-COL-HC09-06_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28813987\t59033610\t4\t1\t0\n",
      "JHY-COL-HC06-16_blood-wgs-ILLUMINA.fmarked\n",
      "2\t89853363\t92325966\t2\t2\t0\n",
      "JHY-COL-HC06-07_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28805860\t59033610\t3\t1\t0\n",
      "HC19-05\n",
      "6\t57369480\t58779898\t5\t3\t2\n",
      "HC19-05\n",
      "Y\t28790954\t59033375\t5\t1\t0\n",
      "JHY-COL-HC06-05_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28805380\t59033610\t3\t1\t0\n",
      "JHY-COL-HC10-09_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28802537\t59033610\t2\t1\t0\n",
      "JHY-COL-HC13-14_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28791164\t59033357\t2\t1\t0\n",
      "JHY-COL-HC02-14_blood-wgs-ILLUMINA.fmarked\n",
      "7\t61054376\t64552765\t5\t3\t2\n",
      "JHY-COL-HC09-08_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28807549\t59033610\t3\t1\t0\n",
      "JHY-COL-HC04-22_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28803998\t59033610\t2\t1\t0\n",
      "JHY-COL-HC10-04_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28800295\t59033610\t2\t1\t0\n",
      "JHY-COL-HC06-04_blood-wgs-ILLUMINA.fmarked\n",
      "2\t90447632\t92325966\t2\t2\t0\n",
      "JHY-COL-HC06-04_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28792268\t59033610\t3\t1\t0\n",
      "JHY-COL-HC05-22_blood-wgs-ILLUMINA.fmarked\n",
      "6\t104786564\t106271749\t2\t2\t0\n",
      "JHY-COL-HC12-16_blood-wgs-ILLUMINA.fmarked\n",
      "10\t96312503\t99839578\t1\t1\t0\n",
      "JHY-COL-HC05-16_blood-wgs-ILLUMINA.fmarked\n",
      "6\t104786564\t106271749\t2\t2\t0\n",
      "JHY-COL-HC05-16_blood-wgs-ILLUMINA.fmarked\n",
      "16\t81487537\t87446368\t1\t1\t0\n",
      "JHY-COL-HC09-16_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28810476\t59033610\t3\t1\t0\n",
      "JHY-COL-HC12-07_blood-wgs-ILLUMINA.fmarked\n",
      "22\t29334848\t51241613\t2\t2\t0\n",
      "JHY-COL-HC13-15_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t23899386\t59033357\t2\t1\t0\n",
      "HC05-02\n",
      "8\t79078613\t80242245\t2\t2\t0\n",
      "JHY-COL-HC06-02_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28788495\t59033610\t3\t1\t0\n",
      "JHY-COL-HC04-08_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28797654\t59033610\t2\t1\t0\n",
      "JHY-COL-HC10-02_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28814425\t59033610\t3\t1\t0\n",
      "JHY-COL-HC13-21_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28790962\t59033357\t2\t1\t0\n",
      "JHY-COL-HC19-08_blood-wgs-ILLUMINA.fmarked\n",
      "6\t57371404\t58779898\t4\t2\t2\n",
      "JHY-COL-HC13-11_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28802591\t59033357\t2\t1\t0\n",
      "JHY-COL-HC19-22_blood-wgs-ILLUMINA.fmarked\n",
      "2\t89880051\t92325966\t3\t2\t1\n",
      "JHY-COL-HC10-10_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t23899234\t59033610\t2\t1\t0\n",
      "JHY-COL-HC04-14_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28812592\t59033610\t3\t1\t0\n",
      "JHY-COL-HC14-18_blood-wgs-ILLUMINA.fmarked\n",
      "7\t61054376\t63233605\t3\t2\t1\n",
      "JHY-COL-HC13-12_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28806847\t59033357\t3\t1\t0\n",
      "JHY-COL-HC08-10_blood-wgs-ILLUMINA.fmarked\n",
      "6\t61932\t29459260\t2\t2\t0\n",
      "JHY-COL-HC04-23_blood-wgs-ILLUMINA.fmarked\n",
      "2\t90125511\t92325966\t2\t2\t0\n",
      "HC16-02\n",
      "9\t44644902\t45769312\t3\t2\t1\n",
      "JHY-COL-HC13-03_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28814425\t59033357\t3\t1\t0\n",
      "JHY-COL-HC16-07_blood-wgs-ILLUMINA.fmarked\n",
      "7\t61054376\t63210479\t5\t3\t2\n",
      "JHY-COL-HC16-04_blood-wgs-ILLUMINA.fmarked\n",
      "2\t90381844\t92325700\t2\t2\t0\n",
      "JHY-COL-HC05-23_blood-wgs-ILLUMINA.fmarked\n",
      "2\t89922171\t92325966\t2\t2\t0\n",
      "HC19-03\n",
      "Y\t28788687\t59033375\t4\t1\t0\n",
      "JHY-COL-HC13-22_blood-wgs-ILLUMINA.fmarked\n",
      "Y\t28797569\t59033357\t2\t1\t0\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Extra to retrieve abn. CN samples\n",
    "id_list = ['HC01','HC02','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC15','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "gender_list = ['M','M','M','M','F','M','F','M','M','F','M','F','F','F','F','F','M','F','F']\n",
    "\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "n=0\n",
    "for i in temp_list:\n",
    "    if \".100kbcov\" not in i and \".gz\" not in i and \".R\" not in i and \".sh\" not in i and \".py\" not in i and '.ipynb' not in i:\n",
    "        if \"qsub\" not in i:\n",
    "            file_list.append(i)\n",
    "            #print i\n",
    "print len(file_list)\n",
    "#sys.exit(1)\n",
    "\n",
    "for input_fn in file_list:\n",
    "    input_file = file(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/%s/%s_segments.txt.clean\" % (input_fn,input_fn))\n",
    "    temp_gender = ''\n",
    "    temp=0\n",
    "    for j in id_list:\n",
    "        if j in input_fn:\n",
    "            temp_gender = gender_list[id_list.index(j)]\n",
    "            break\n",
    "    if temp== 1:\n",
    "        input_file.close()\n",
    "        continue\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if temp_gender == 'F':\n",
    "            if input_split[4] !='1' or input_split[5] != '1':\n",
    "                print input_fn\n",
    "                print input_line\n",
    "                \n",
    "            \n",
    "        elif temp_gender == 'M':\n",
    "            if input_split[0] == 'X' or input_split[0] == 'Y':\n",
    "                if input_split[3] != '1':\n",
    "                    print input_fn\n",
    "                    print input_line\n",
    "            else:\n",
    "                if input_split[4] !='1' or input_split[5] != '1':\n",
    "                    print input_fn\n",
    "                    print input_line                \n",
    "            \n",
    "        else:\n",
    "            print 'UNEXPECTED'\n",
    "        input_line = input_file.readline().strip()\n",
    "    input_file.close()\n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "405\n",
      "HC03-12\n",
      "\"Y\"\t2932746\t9908828\t0.0377839274524644\t252\t0.0907239797140878\t0.228412142651727\t1804\t0.330629526664721\t1\tNA\tNA\t-15.8484793895773\n",
      "HC03-12\n",
      "\"Y\"\t13869877\t21155023\t0.0365654802990336\t114\t0.0884266331245188\t0.169359215904889\t989\t0.228355900471977\t1\tNA\tNA\t-19.233847560011\n",
      "HC03-12\n",
      "\"Y\"\t22225604\t23901231\t0.0222963288521193\t91\t0.0897697502425403\t0.0520302137619866\t444\t0.0256115809532607\t1\tNA\tNA\t-30.9764758745962\n",
      "HC03-12\n",
      "\"Y\"\t28805301\t58905551\t0.158932282002784\t181\t0.177278277811797\t0.0217065356360928\t564\t0.0266487826290507\t1\tNA\tNA\t-40.9572028945365\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Extra to retrieve abn. CN samples\n",
    "id_list = ['HC01','HC02','HC03','HC04','HC05','HC06','HC08','HC09','HC10','HC12','HC13','HC14','HC15','HC16','HC17','HC18','HC19','HC20','HC21'] #'HC15'\n",
    "gender_list = ['M','M','M','M','F','M','F','M','M','F','M','F','F','F','F','F','M','F','F']\n",
    "\n",
    "\n",
    "import os,sys\n",
    "temp_list = os.listdir(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/\")\n",
    "file_list = []\n",
    "pre_list = []\n",
    "n=0\n",
    "for i in temp_list:\n",
    "    if \".100kbcov\" not in i and \".gz\" not in i and \".R\" not in i and \".sh\" not in i and \".py\" not in i and '.ipynb' not in i:\n",
    "        if \"qsub\" not in i:\n",
    "            file_list.append(i)\n",
    "            #print i\n",
    "print len(file_list)\n",
    "#sys.exit(1)\n",
    "\n",
    "for input_fn in file_list:\n",
    "    input_file = file(\"/home/users/jhyouk/14_colon_LINE1/04_sequenza_gender/%s/%s_segments.txt\" % (input_fn,input_fn))\n",
    "    temp_gender = ''\n",
    "    temp=0\n",
    "    for j in id_list:\n",
    "        if j in input_fn:\n",
    "            temp_gender = gender_list[id_list.index(j)]\n",
    "            break\n",
    "    if temp== 1:\n",
    "        input_file.close()\n",
    "        continue\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if int(input_split[2]) - int(input_split[1]) > 100000:\n",
    "            if float(input_split[6]) < 0.3:\n",
    "                print input_fn\n",
    "                print input_line\n",
    "        input_line = input_file.readline().strip()\n",
    "    input_file.close()\n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
